lib/ansible/vars/reserved.py

```diff
@@ -76,8 +75,11 @@ def warn_if_reserved(myvars: c.Iterable[str], additional: c.Iterable[str] | None
     varnames = set(myvars)
     varnames.discard('vars')  # we add this one internally, so safe to ignore
 
-    for varname in varnames.intersection(reserved):
-        display.warning(f'Found variable using reserved name {varname!r}.')
+    if conflicts := varnames.intersection(reserved):
+        # Ensure the varname used for obj is the tagged one from myvars and not the untagged one from reserved.
+        # This can occur because tags do not affect value equality, and intersection can return values from either the left or right side.
+        for varname in (name for name in myvars if name in conflicts):
```

Comment:
isn't `confllicts` the only thing we need to iterate? if it is there its already in myvars, there is no need to check again

Reply:
Nvmd